src/slurm_plugin/slurm_resources.py

```diff
@@ -255,8 +261,8 @@ def needs_reset_when_inactive(self):
 
     def get_compute_resource_name(self):
         """Get instance name of given node."""
-        _, _, compute_resource_name = parse_nodename(self.name)
-        return compute_resource_name
+        queue_name, _, compute_resource_name = parse_nodename(self.name)
+        return queue_name + "-" + compute_resource_name
```

Comment:
I don't think it's correct to have a method that is supposed to return the cr name to return something different. Also there might be other places of the code where this method is used.

One thing we could do so that the parsing is done only once and caller can access individual fields is to call the parse_nodename in the node `__init__` method and assign to class attributes the returned values (e.g. `self.queue_name, self._node_type, self.compute_resource_name = parse_nodename(...)`

Reply:
This is a good idea, changed to parse_nodename in the node __init__ method and removed this function